### PR TITLE
Gallery - filter and view

### PR DESCRIPTION
**Pull request by by-co**
## Feature Name
Gallery page for works, sort and filter functionality
[Asana](https://app.asana.com/0/860297512787618/881723799904621)
### Related PRs
None
### Migrations
Rename columns in `Work`: fef74e0a1

Purpose is so that the enums become the single source of truth for filter categories and filter items. We can directly use the strings on the front end app.

```
- "media" => "materials"
  - Short text field, e.g. 'OIL ON CANVAS'
- "work_type" => "media"
  - One of 5 options, e.g. 'PAINTING'
- "status" => "availability"
  - One of 3 options, e.g. 'SOLD'
```
### Tests Performed, Edge Cases
Can handle:
- uncheck all, test again
- multiple items within same category

*Need to rethink*:
- What happens if the user selects "Available" + "Painting"?
  - Right now we'll show the *least selective* filter, e.g. 'available'
### Screenshots

screen shot 2018-10-27 at 8 38 42 pm

CC: @clairelin135

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -2,6 +2,7 @@
 <html>
   <head>
     <title>SFAI</title>
+    <link rel="stylesheet" href="https://unpkg.com/tachyons@4.10.0/css/tachyons.min.css"/>
```
[the remark this answers is not in the document]
Comment (by-co) [reply app/views/layouts/application.html.erb]: I think tachyons.min means the entire base stylesheet is [minified](https://blog.stackpath.com/glossary/minification/)?

However, I remember you mentioned that you got the tachyons-sass library to work, so I can remove this library if we're using a custom tachyons library now

**Comment by lbkchen** [comment]
This is a good PR

File: app/javascript/components/Works.jsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      );
+  };
+
+  render() {
+    const { filters, works } = this.state;
+
+    return (
+      <div className="ma4">
+        <div className="flex bg-grey">
+          <Filters filters={filters} onFiltersChange={this.handleFilters} />
```
[the remark this answers is not in the document]
Comment (by-co) [reply app/javascript/components/Works.jsx]: Great suggestion, implemented it!

File: config/routes.rb [hunk 3]
```diff
@@ -4,17 +4,20 @@
   devise_for :buyers
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
   root 'pages#home'
```
[the remark this answers is not in the document]
Comment (by-co) [reply config/routes.rb]: I think eventually we want to have a landing page with some splashy images / copywriting, though 😮

File: config/routes.rb [hunk 4]
```diff
@@ -4,17 +4,19 @@
...
   get 'artists/:id' => 'artists#show', as: :artistid
-  get '/artists/', to: 'artists#all_artists'
+  get 'artists/', to: 'artists#all_artists'
+  get 'works/' => 'works#index'
+  get 'works/categories' => 'works#get_work_category_enums'
   get 'buyers/:id' => 'buyers#show', as: :buyerid
 
   namespace :api, defaults: { format: :json } do
-    resources :works, :only => [:show, :create, :update, :destroy]
+    resources :works, :only => [:index, :create, :update, :destroy]
```
Comment (clairelin135) [inline config/routes.rb]: We should restructure the `works/:id/show` method to be part of this line. The structure of the route should instead be `works/show/:id`, and then we can include `show` in this resource mapping the way that it's done for  artists below

File: config/routes.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (clairelin135) [reply config/routes.rb]: I think that until sfai provides this content, we should root to the index page since this is essentially an art gallery platform.

File: app/javascript/components/Works.jsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+      ? APIRoutes.works.filtered_works(searchParams)
+      : APIRoutes.works.index;
+
+    Requester.get(
+      works_route).then(
+        response => {
+          this.setState({ works: response });
+        },
+        response => {
+          console.err(response);
```
Comment (clairelin135) [inline app/javascript/components/Works.jsx]: also console.err is not a function, i think its console.error

**Comment by clairelin135** [comment]
WOOOOOOOOOOOOOOOOO
